src/mapnik_map.cpp

```diff
@@ -936,6 +937,25 @@ NAN_METHOD(Map::add_layer) {
     return;
 }
 
+/**
+ * Remove layer from this map
+ *
+ * @memberof Map
+ * @instance
+ * @name remove_layer
+ * @param {mapnik.Layer} new layer
+ */
+NAN_METHOD(Map::remove_layer) {
+    if (!info[0]->IsNumber()) {
+        Nan::ThrowTypeError("index must be number");
+        return;
+    }
+
+    Map* m = Nan::ObjectWrap::Unwrap<Map>(info.Holder());
+    m->map_->remove_layer(info[0]->IntegerValue());
```

Comment:
What would happen if layer index is outside of valid range e.g `<0` or `>num_layers` ? 'remove_layer' calls `erase(begin+ index)`, I think we need an extra validation logic to check layer index is valid, what do think?

Reply:
I've improved arguments validation.

Reply:
Great, thanks.